src/topology/paracompact.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Reid Barton, Yury Kudryashov
+-/
+import topology.subset_properties
+import topology.separation
+import data.option.basic
+
+/-!
+# Paracompact topological spaces
+
+In this file we define a `paracompact_space` and provide two instances:
+
+- a compact space is paracompact;
+- a locally compact sigma compact Hausdorff space is paracompact.
+
+We also prove that every paracompact Hausdorff space is normal. This statement is not an instance
+to avoid loops in the instance graph.
+
+## TODO
+
+Define partition of unity
+
+## Tags
+
+compact space, paracompact space
+-/
+
+open set filter function
+open_locale filter topological_space
+
+universes u v
+
+/-- A topological space is called paracompact, if every open covering of this space admits
+a locally finite refinement. -/
+class paracompact_space (X : Type u) [topological_space X] : Prop :=
+(locally_finite_refinement :
+  ∀ (S : set (set X)) (ho : ∀ s ∈ S, is_open s) (hc : ⋃₀ S = univ),
+  ∃ (α : Type u) (t : α → set X) (ho : ∀ a, is_open (t a)) (hc : (⋃ a, t a) = univ),
+    locally_finite t ∧ ∀ a, ∃ s ∈ S, t a ⊆ s)
```

Comment:
What motivates using `Type u` as the universe for `α` here instead of `Type` or `Sort v`? Is the argument simply "no one cares about universe polymorphism"? If so, a brief comment to that effect would probably be a good idea.

I assume using something like the below is undesirable?

```suggestion
  ∃ (t : set (set X)) (ho : ∀ a ∈ t, is_open a) (hc : (⋃ a ∈ t, a) = univ),
    locally_finite sorry ∧ ∀ a ∈ t, ∃ s ∈ S, a ⊆ s)
```

(disclaimer: I know exactly none of the maths here)

Reply:
`Type u` is here to avoid an extra universe parameter in the typeclass definition. While it is possible to write the definition using `set (set X)`, **all** instances construct this set as the range of some function, so it's easier to state the definition this way. The next lemma deals starts with a covering indexed by `ι : Type*` and returns a refinement with the same index type.